### PR TITLE
Add changes from 2225-2227 amazon-FreeRTOS

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
Bring in the changes from PRs https://github.com/aws/amazon-freertos/pull/2225, https://github.com/aws/amazon-freertos/pull/2226 and https://github.com/aws/amazon-freertos/pull/2227 to https://github.com/FreeRTOS/FreeRTOS.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/include/FreeRTOSIPConfigDefaults.h [hunk 1]
```diff
@@ -314,7 +314,25 @@ from the FreeRTOSIPConfig.h configuration header file. */
...
+#ifndef ipconfigUDP_PASS_ZERO_CHECKSUM_PACKETS
+	#define ipconfigUDP_PASS_ZERO_CHECKSUM_PACKETS	0
+#endif
+
+
+#ifndef ipconfigIP_PASS_PACKETS_WITH_IP_OPTIONS
+	#define ipconfigIP_PASS_PACKETS_WITH_IP_OPTIONS	1
+#endif
+
+#ifndef ipconfigUDP_PASS_ZERO_CHECKSUM_PACKETS
```
Comment (aggarw13) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/include/FreeRTOSIPConfigDefaults.h:334]: Seems like the default definition logic for `ipconfigIP_PASS_PACKETS_WITH_IP_OPTIONS` and `ipconfigUDP_PASS_ZERO_CHECKSUM_PACKETS` macros is repeated

Comment (aggarw13) [reply FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/include/FreeRTOSIPConfigDefaults.h:334]: Also, is there any documentation available for these macros for the customer to understand their utility?

Comment (gkwicker) [reply FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/include/FreeRTOSIPConfigDefaults.h:334]: I asked for that in a previous version of this code @aggarw13 but the author's opinion was that they are self-documenting.

**Comment by aggarw13** [comment]
Approved the PR as the files match their equivalent on `master` branch of `aws/amazon-freertos` repo

**Pull request merged**